tests/v2/validation/clustertemplates/provisioning/provisioning_clustertemplate_test.go

```diff
@@ -0,0 +1,236 @@
+
+package clustertemplates
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/rancher/rancher/tests/v2/actions/provisioning"
+	"github.com/rancher/rancher/tests/v2/actions/provisioninginput"
+	templates "github.com/rancher/rancher/tests/v2/validation/clustertemplates"
+	"github.com/rancher/shepherd/clients/rancher"
+	mgmt "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	extensionscluster "github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/clusters/kubernetesversions"
+	"github.com/rancher/shepherd/extensions/settings"
+	"github.com/rancher/shepherd/extensions/users"
+	"github.com/rancher/shepherd/pkg/config"
+
+	"github.com/rancher/shepherd/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ClusterTemplateProvisioningTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	provisioningConfig *provisioninginput.Config
+}
+
+func (ct *ClusterTemplateProvisioningTestSuite) TearDownSuite() {
+	ct.session.Cleanup()
+}
+
+func (ct *ClusterTemplateProvisioningTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	ct.session = testSession
+
+	ct.provisioningConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, ct.provisioningConfig)
+
+	client, err := rancher.NewClient("", testSession) //ct.client
+	require.NoError(ct.T(), err)
+
+	ct.client = client
+
+	if ct.provisioningConfig.RKE1KubernetesVersions == nil {
+		rke1Versions, err := kubernetesversions.ListRKE1AllVersions(ct.client)
+		require.NoError(ct.T(), err)
+
+		ct.provisioningConfig.RKE1KubernetesVersions = []string{rke1Versions[len(rke1Versions)-1]}
+	}
+
+	if ct.provisioningConfig.CNIs == nil {
+		ct.provisioningConfig.CNIs = []string{templates.CniCalico}
+	}
+
+}
+
+func (ct *ClusterTemplateProvisioningTestSuite) TestProvisioningRKE1ClusterWithClusterTemplate() {
+	log.Info("Create an rke template and creating a downstream node driver with the rke template.")
+
+	clusterTemplate, err := templates.CreateRkeTemplate(ct.client, nil)
+	require.NoError(ct.T(), err)
+
+	templateRevisonConfig := mgmt.ClusterTemplateRevision{
+		ClusterConfig: &mgmt.ClusterSpecBase{RancherKubernetesEngineConfig: &mgmt.RancherKubernetesEngineConfig{
+			Network: &mgmt.NetworkConfig{Plugin: ct.provisioningConfig.CNIs[0]},
+			Version: ct.provisioningConfig.RKE1KubernetesVersions[0],
+		},
+		},
+	}
+
+	clusterTemplateRevision, err := templates.CreateRkeTemplateRevision(ct.client, templateRevisonConfig, clusterTemplate.ID)
+	require.NoError(ct.T(), err)
+
+	rke1Cluster := templates.UpdateClusterTemplateClusterConfig(clusterTemplate.ID, clusterTemplateRevision.ID, nil)
+
+	clusterObj, clusterConfig, err := templates.CreateRke1ClusterWithTemplateAndVerify(ct.client, rke1Cluster, ct.provisioningConfig)
```

Comment:
I think to avoid potential flakiness, we should add a verify function on the clusterTemplate(and/or revision)
as the create isn't waiting on the clusterTemplate to come up, nor its revision. So it would be safe to add a verifyClusterTemplate func that waits on the resource to be ready. That could be as simple as checking that a given part of the object is there when you get it from v3 api. But we should have something that checks it is there before creating from it.

Reply:
I don't feel too strongly either way - but even if it might add some overhead, its almost always worth it to validate that the resource we asked rancher to create is actually ready - we almost always run into an issue with it down the road and will lead to some confusion if you don't validate it before creating with it. 

i.e.
user runs the test
the test fails to create the cluster
user looks at the setup, and the template is ready to use
deploys with template, everything deployed fine

its adding work for us later to figure out in a flaky situation vs. adding something simple now to save us the headache later.

Reply:
ok, I added a check to verify the status on the created object.